internal/pkg/plugin/githubintegrations/githubintegrations.go

```diff
@@ -0,0 +1,255 @@
+	wsFiles := make([]string, 0)
+	for _, w := range workflows {
+		wsFiles = append(wsFiles, w.workflowFileName)
+	}
+	fmt.Printf("Workflow files: %v", wsFiles)
+
+	_, dirContent, resp, err := gi.client.Repositories.GetContents(
+		gi.ctx,
+		gi.options.Owner,
+		gi.options.Repo,
+		".github/workflows",
+		&github.RepositoryContentGetOptions{},
+	)
+
+	// error reason is not 404
+	if err != nil && !strings.Contains(err.Error(), "404") {
+		log.Printf("GetContents failed with error: %s", err)
+		return nil, err
+	}
+	// StatusCode == 404
+	if resp.StatusCode == http.StatusNotFound {
+		log.Printf("GetContents return with status code 404")
+		retMap := mapz.FillMapWithStrAndError(wsFiles, fmt.Errorf("not found"))
+		return retMap, nil
+	}
+	// StatusCode != 200
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("got some error is not expected: %s", resp.Status)
+	}
+	// StatusCode == 200
+	log.Printf("GetContents return with status code 200")
+	var filesInRemoteDir = make([]string, 0)
+	for _, f := range dirContent {
+		log.Printf("Found remote file: %s", f.GetName())
+		filesInRemoteDir = append(filesInRemoteDir, f.GetName())
+	}
+
+	lostFiles := slicez.SliceInSliceStr(wsFiles, filesInRemoteDir)
+	// all files exist
+	if len(lostFiles) == 0 {
+		log.Println("All files exist")
+		retMap := mapz.FillMapWithStrAndError(wsFiles, nil)
+		return retMap, nil
+	}
+	// some files lost
+	log.Println("Some files lost")
+	retMap := mapz.FillMapWithStrAndError(wsFiles, nil)
+	for _, f := range lostFiles {
+		log.Printf("Lost file: %s", f)
+		retMap[f] = fmt.Errorf("not found")
+	}
+	return retMap, nil
+}
+
+// getFileSHA will try to collect the SHA hash value of the file, then return it. the return values will be:
+// 1. If file exists without error -> string(SHA), nil
+// 2. If some errors occurred -> return "", err
+// 3. If file not found without error -> return "", nil
+func (gi *GithubIntegrations) getFileSHA(filename string) (string, error) {
+	content, _, resp, err := gi.client.Repositories.GetContents(
+		gi.ctx,
+		gi.options.Owner,
+		gi.options.Repo,
+		generateGitHubWorkflowFileByName(filename),
+		&github.RepositoryContentGetOptions{},
+	)
+
+	// error reason is not 404
+	if err != nil && !strings.Contains(err.Error(), "404") {
+		return "", err
+	}
+
+	// error reason is 404
+	if resp.StatusCode == http.StatusNotFound {
+		return "", nil
+	}
+
+	// no error occurred
+	if resp.StatusCode == http.StatusOK {
+		return *content.SHA, nil
+	}
+	return "", fmt.Errorf("got some error is not expected")
+}
+
+// renderTemplate render the github actions template with config.yaml
+func (gi *GithubIntegrations) renderTemplate(workflow *Workflow) error {
+	var jobs Jobs
+	err := mapstructure.Decode(gi.options.Jobs, &jobs)
+	if err != nil {
+		return err
+	}
+	//if use default {{.}}, it will confict (github actions vars also use them)
+	t, err := template.New("githubactions").Delims("[[", "]]").Parse(workflow.workflowContent)
```

Comment:
template name needs to be updated. Do a grep over this directory and make sure no "githubactions" should exist.